components/Ink/src/MDCSimpleInkView.m

```diff
@@ -0,0 +1,93 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCSimpleInkView.h"
+#import "MDCSimpleInkGestureRecognizer.h"
+#import "private/MDCSimpleInkLayer.h"
+
+@interface MDCSimpleInkView ()
+
+@property(nonatomic, strong) MDCSimpleInkLayer *inkLayer;
+
+@end
+
+@implementation MDCSimpleInkView
+
+- (instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    _inkColor = [UIColor colorWithWhite:0 alpha:0.08f];
+  }
+  return self;
+}
+
+- (void)setFrame:(CGRect)frame {
+  [super setFrame:frame];
+  self.inkLayer.frame = frame;
+}
+
+- (void)startInkAtPoint:(CGPoint)point
+             completion:(MDCSimpleInkCompletionBlock)completionBlock {
+  self.inkLayer = [MDCSimpleInkLayer layer];
+  self.inkLayer.inkColor = self.inkColor;
+  self.inkLayer.completionBlock = completionBlock;
+  self.inkLayer.opacity = 0;
+  self.inkLayer.frame = self.bounds;
+  [self.layer addSublayer:self.inkLayer];
```

Comment:
What are the benefits of adding as a sublayer over replacing the existing CALayer with our CAInkLayer?
Should we combine our CAInkLayer and our CAShadowLayer into a single CALayer?

Reply:
MDCSimpleInkLayer as a sublayer makes it easier to animate scale and position in relation to its parent. When MDCSimpleInkView's CALayer is modified, it is a bit more challenging to animate scale and position correctly. Also, because of the scaling and position animations, it won't make as much sense to combine the shadow and ink layers, as doing so will increase the complexity of the layer considerable, and will likely make it necessary to create multiple sublayers to separate the ink animation from the shadow transformations.

Reply:
The layer should be removed when the animation ends. Layer removal used to be in the ink layer class, but I've moved this into the ink view with an animation end protocol method to conform to where the ink layer will be removed.